From 5982f6ceb9f4775816c9472c78afc263a6172c23 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Mon, 4 Jun 2018 10:05:37 +0200
Subject: [PATCH 17/18] PhoneWindowManager: Improve home button wake haptic
 feedback handling

* This fixes an issue where haptic feedback is used
  when screen is off and home button wake is disabled.

Change-Id: I7ac4c00598cedf7f174dc99629f55dc7b74b0d2a
---
 .../java/com/android/server/policy/PhoneWindowManager.java | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 6b14632d5b7..c91b416d6e8 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -6743,8 +6743,11 @@ public class PhoneWindowManager implements WindowManagerPolicy {
             }
 
             case KeyEvent.KEYCODE_HOME:
-                if (down && !interactive && mHomeWakeScreen) {
-                    isWakeKey = true;
+                if (down && !interactive) {
+                    isWakeKey = mHomeWakeScreen;
+                    if (!isWakeKey) {
+                        useHapticFeedback = false;
+                    }
                 }
                 break;
 
-- 
2.17.1

